lib/client.ts

```diff
@@ -75,12 +76,30 @@ class Client {
         }
     }
     async connect() {
+        const key = `${this.relayUrl}:${this.mnemonics}:${this.keypairType}`;
+        if (Client.connections.has(key)) {
+            return Client.connections.get(key);
```

Comment:
I think this code shouldn't be here in the connect method. it should be in the constructor. as if a user created an instance. then calls connect on this instance, it will work fine. but if he creates another instance, the connection won't be in the second instance. so he has to override the second instance with the return of the connect method after calling it. and IMO, this a bad usage. user should create the instance and it should be working without caring about if it has a connection or not and has to override or not